packages/runtime/datastore/src/dataStoreRuntime.ts

```diff
@@ -123,8 +123,7 @@ IFluidDataStoreChannel, IFluidDataStoreRuntime, IFluidHandleContext {
     }
 
     public get clientDetails(): IClientDetails {
-        // back-compat 0.38 - clientDetails is added to IFluidDataStoreContext in 0.38.
-        return this.dataStoreContext.clientDetails ?? this.dataStoreContext.containerRuntime.clientDetails;
```

Comment:
Can we remove IContainerRuntime(Base).clientDetails ? DataStoreContext works with concrete ContainerRuntime class, so it does not care.